stabilizer/src/main/java/com/hazelcast/stabilizer/provisioner/Provisioner.java

```diff
@@ -391,25 +393,29 @@ public void terminate(int count) {
 
         long durationSeconds = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis() - startMs);
         echo("Duration: " + secondsToHuman(durationSeconds));
-        echoImportant("Finished terminating %s %s machines, %s machines remaining.", count, props.get("CLOUD_PROVIDER"),
-                addresses.size());
+        echoImportant("Terminated %s of %s, remaining=%s", destroyedCount, count, addresses.size());
+
+        if(destroyedCount!=count){
+            throw new IllegalStateException("Terminated "+destroyedCount+" of "+count);
```

Comment:
a nicer explanation would be appreciated. such as possible hints what might go wrong. (Harakiri? anything else?)